### PR TITLE
Add external scheduler metrics

### PR DESCRIPTION
**Pull request by ykmr1224**
### Description
- Add external scheduler metrics
  - externalScheduler.{action}.count

Tested in my dev environment:
externalScheduler metrics

### Related Issues
n/a

### Check List
- [ ] Updated documentation (docs/ppl-lang/README.md)
- [ ] Implemented unit tests
- [ ] Implemented tests for combination with other commands
- [ ] New added source code should include a copyright header
- [x] Commits are signed per the DCO using `--signoff`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/sql/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by noCharger — COMMENTED** [review]
these already exist in flint-spark-integration/src/main/scala/org/opensearch/flint/spark/refresh/util/RefreshMetricsHelper.scala

**Comment by ykmr1224** [comment]
> these already exist in flint-spark-integration/src/main/scala/org/opensearch/flint/spark/refresh/util/RefreshMetricsHelper.scala

It is intended to track each action. Is the existing metric track them? (I though it is tracking success/failure/latency for exernal scheduler action)

**Comment by noCharger** [comment]
> > these already exist in flint-spark-integration/src/main/scala/org/opensearch/flint/spark/refresh/util/RefreshMetricsHelper.scala
> 
> It is intended to track each action. Is the existing metric track them? (I though it is tracking success/failure/latency for exernal scheduler action)

what's the benefit of adding action? it's mainly opensearch CRUD

**Comment by ykmr1224** [comment]
> > > these already exist in flint-spark-integration/src/main/scala/org/opensearch/flint/spark/refresh/util/RefreshMetricsHelper.scala
> > 
> > 
> > It is intended to track each action. Is the existing metric track them? (I though it is tracking success/failure/latency for exernal scheduler action)
> 
> what's the benefit of adding action? it's mainly opensearch CRUD

It would show how often user add/modify/delete manual refresh. If we can easily know it with other metrics, we can skip it.

File: flint-spark-integration/src/main/scala/org/opensearch/flint/spark/scheduler/FlintSparkJobExternalSchedulingService.scala [hunk 1]
```diff
@@ -83,8 +84,16 @@ class FlintSparkJobExternalSchedulingService(
...
 
       None // Return None for all cases
     }
   }
+
+  private def addExternalSchedulerMetrics(action: AsyncQuerySchedulerAction): Unit = {
+    val actionName = action.name().toLowerCase()
+    MetricsUtil.addHistoricGauge(
+      MetricConstants.EXTERNAL_SCHEDULER_METRIC_PREFIX + actionName + ".count",
+      1)
```
Comment (noCharger) [inline flint-spark-integration/src/main/scala/org/opensearch/flint/spark/scheduler/FlintSparkJobExternalSchedulingService.scala:97]: Will HistoricGauge with value one aggregated on client side?

File: flint-spark-integration/src/main/scala/org/opensearch/flint/spark/scheduler/FlintSparkJobExternalSchedulingService.scala [hunk 2]
```diff
@@ -83,8 +84,16 @@ class FlintSparkJobExternalSchedulingService(
         case AsyncQuerySchedulerAction.REMOVE => flintAsyncQueryScheduler.removeJob(request)
         case _ => throw new IllegalArgumentException(s"Unsupported action: $action")
       }
+      addExternalSchedulerMetrics(action)
```
Comment (noCharger) [inline flint-spark-integration/src/main/scala/org/opensearch/flint/spark/scheduler/FlintSparkJobExternalSchedulingService.scala:87]: can we reuse withMetrics by adding the action in prefix?

**Review by noCharger — APPROVED** [review]

**Pull request merged**